flask_multipass/core.py

```diff
@@ -145,6 +149,11 @@ def validate_next_url(self, url):
         If you override this and want to allow more hosts, make sure to use
         a whitelist of trusted hosts to avoid creating an open redirector.
         """
+        # next_url comes as URL param, so it can have newline chars (as %0a and %0d).
+        # In redirect response it goes into header (Location).
+        # Werkzeug doesn't tolerate newline chars in header and raises ValueError.
+        if _newline_re.search(url) is not None:
```

Comment:
Not sure this needs a regex... a simple `'\n' in url or '\r' in url` does the job just fine.

I'd also include `\0` since that also makes no sense there (and might trigger an error on the werkzeug side).

Reply:
ah i see you took the code from werkzeug. fine for me to leave it as-is then.

Another - maybe nicer - option would be to try constructing a Headers object and checking if it raises a ValueError when converted to string. That way you can reuse the werkzeug logic w/o replicating it.